proposals/tf_utf8_identifiers/README.md

```diff
@@ -0,0 +1,443 @@
+- **Identifiers:** The identifying names given to USD schema types and properties, largely with the expectation that they can be syntactically combined to be compiled into APIs
+
+The current implementation treats these the same, and restricts both to the rules given above largely due to their use in compiling schema representations that must be compliant with C++ and Python.  However, there is little need to restrict prim names in the same way, and often the source of these names in industries such as AECO and manufacturing come from systems that are permissive in the way these names are represented (e.g. names in non-Latin based languages, spaces, all numerical names, etc.).  Below we discuss one possible approach to distinguish between the two concepts and what would need to change to support doing so.
+
+# Considerations for Implementation
+
+For the proposed implementation, it was decided to use the following conventions:
+
+- Standardize on UTF-8 encoding of Unicode characters
+- Implement as much as possible in USD core without relying on external libraries
+- Consolidate validation routines down to the `Tf` module
+
+Modifying these code paths to support Unicode requires standardizing on a character encoding and expanding the identifier rules to accept Unicode strings. Unicode strings can be encoded in various forms, some of the most popular being:
+
+- code points
+- UTF-8
+- UTF-16 / UTF-32
+
+In evaluating options to support, raw code points rely largely on the support of code pages and individual system locales.  UTF-x encodings are already populate for standardized interchange of Unicode text and in particular UTF-8 encoding is widespread within the community.  UTF-8 is a variable length encoding that is backward compatible with existing ASCII encodings, meaning that current usd files, code, etc. that rely on ASCII strings should work without any changes.  The reader is referred to [UTF-8 Everywhere](http://utf8everywhere.org/) as a summary of why UTF-8 was chosen as the preferred encoding over UTF-16 / UTF-32.
+
+Building USD is already a complex endevor.  While there are existing libraries that do provide APIs for working with Unicode character sequences, like [ICU](https://icu.unicode.org/), using these libraries would necessitate the acquisition and distribution of this library with the core USD libraries.  Efforts are already underway to eliminate dependencies (e.g., Boost), so we felt that a small enough subset of Unicode string processing could be implemented in USD core without the need for external libraries.  In practice, the choice of collation algorithm is the primary driver of whether or not to use an external library, so decisions on doing so should be deferred until after understanding the collation algorithm that should be used with the addition of Unicode identifiers (see [Identifier Collation](#identifier-collation) for the different proposed options).
```

Comment:
Should this be "the choice of _sorting_ algorithm", "understanding the _sorting_ algorithm", and a link to "Identifier Sorting"